source/shared/cpp/ObjectModel/MarkDownParser.cpp

```diff
@@ -0,0 +1,430 @@
+#include <vector>
+#include <iomanip>
+#include <iostream>
+#include "MarkDownParser.h"
+
+using namespace AdaptiveCards;
+
+MarkDownParser::MarkDownParser (const std::string &txt) : m_currentDelimiterType (Init), 
+    m_currentWordIndex(0), m_lookBehind(Init), m_delimiterCnts(0), m_EmphasisState (EmphasisNotDetected), 
+    m_checkLookAhead(false), m_checkIntraWord(false), m_isRightEmphasisDetecting(false), m_text(txt), 
+    m_textBegin(m_text.begin()), m_curPos(m_text.begin()), m_textEnd(m_text.end()), m_LeftEmphasisDetecting(0),
+    m_leftLookUpTable (), m_rightLookUpTable (), m_tokenizedString(1, "")
+{
+}
+
+int MarkDownParser::adjustDelimsCntsAndMetaData(int leftOver, std::list<Emphasis>::iterator left_itr, 
```

Comment:
>adjustDelimsCntsAndMetaData [](start = 20, length = 27)

Could you please add a function comment explaining this method?

Reply:
sure

---
In reply to: [154402126](https://github.com/Microsoft/AdaptiveCards/pull/1003#discussion_r154402126) [](ancestors = 154402126)